pkg/admission/mutator/shoot.go

```diff
@@ -112,15 +112,8 @@ func (s *shoot) Mutate(ctx context.Context, new, old client.Object) error {
 			}
 		}
 
-		if networkConfig.Overlay != nil && !networkConfig.Overlay.Enabled {
-			networkConfig.SnatToUpstreamDNS = &calicov1alpha1.SnatToUpstreamDNS{Enabled: true}
-		} else {
-			networkConfig.SnatToUpstreamDNS = &calicov1alpha1.SnatToUpstreamDNS{Enabled: false}
-		}
```

Comment:
Can't we summarize the change (line 92) like so:

```
overlay := &calicov1alpha1.Overlay{Enabled: false}
if EnableOverlayAsDefaultForCalico {
    overlay := &calicov1alpha1.Overlay{Enabled: true}
}
if oldShoot == nil && networkConfig.Overlay == nil {
    networkConfig.Overlay = overlay
}
...
```

and skip the post processing part ? Even more because the as it is right now it is not only enabling overlay "as default" but it applies it in all cases if I am not mistaken.

Reply:
It's not applying it in all cases, when the overlay field is not set for an old cluster, it will also not set it now. But we can change the order.

Reply:
With this change now as proposed we don't change it when the overlay field is already set for a cluster. We only change the overlay field value for new clusters or when the overlay field gets removed.

Reply:
@DockToFuture Can the overlay ever be removed ? Wouldn't it fall under: 

```
if oldShoot != nil && networkConfig.Overlay == nil
```

Reply:
Why would we want to remove this field once it was set? It could either be enabled or disabled.